src/appleseed/renderer/modeling/aov/pixelvariationaov.cpp

```diff
@@ -73,50 +75,13 @@ namespace
 
         void post_process_image(const Frame& frame) override
         {
-            static const Color3f Blue(0.0f, 0.0f, 1.0f);
-            static const Color3f Red(1.0f, 0.0f, 0.0f);
-
             const AABB2u& crop_window = frame.get_crop_window();
+            ColorMap color_map;
+            color_map.set_palette_from_array(InfernoColorMap, countof(InfernoColorMap) / 3);
 
-            // Find the maximum variation.
-            float max_variation = 0.0f;
-
-            for (size_t y = crop_window.min.y; y <= crop_window.max.y; ++y)
-            {
-                for (size_t x = crop_window.min.x; x <= crop_window.max.x; ++x)
-                {
-                    Color3f color;
-                    m_image->get_pixel(x, y, color);
-                    max_variation = max(color[0], max_variation);
-                }
-            }
-
-            // Normalize if a maximum was found.
-            if (max_variation != 0.0f)
-            {
-                for (size_t y = crop_window.min.y; y <= crop_window.max.y; ++y)
-                {
-                    for (size_t x = crop_window.min.x; x <= crop_window.max.x; ++x)
-                    {
-                        Color3f color;
-                        m_image->get_pixel(x, y, color);
-
-                        const float c = fit(color[0], 0.0f, max_variation, 0.0f, 1.0f);
-                        assert(c >= 0.0f && c <= 1.0f);
-                        m_image->set_pixel(x, y, lerp(Blue, Red, c));
-                    }
-                }
-            }
-            else
-            {
-                for (size_t y = crop_window.min.y; y <= crop_window.max.y; ++y)
-                {
-                    for (size_t x = crop_window.min.x; x <= crop_window.max.x; ++x)
-                    {
-                        m_image->set_pixel(x, y, Blue);
-                    }
-                }
-            }
+            float min_var, max_var;
+            color_map.find_min_max_red_channel(&frame.image(), crop_window, min_var, max_var);
```

Comment:
You're using `&frame.image()` here and `m_image` on the next line.

Reply:
Is there a problem? As far as I can tell both points to the same image object.
Regardless, I changed them both to `*m_image` (function takes in ref now) for consistency's sake.

Reply:
No problem, just a lack of consistency :)